core/types/celo_transaction_signing_forks.go

```diff
@@ -64,6 +64,8 @@ func (c *cel2) txFuncs(tx *Transaction) *txFuncs {
 	switch {
 	case t == LegacyTxType && tx.IsCeloLegacy():
 		return deprecatedTxFuncs
+	case t == CeloDynamicFeeTxType:
+		return deprecatedTxFuncs
 	case t == CeloDenominatedTxType:
 		return deprecatedTxFuncs
```

Comment:
Shouldn't I only see CIP-64 here?

Instead I see
* Celo legacy (not supported anymore)
* CIP-42 (not supported anymore, so you correctly suggest removing it)
* CIP-66 (not supported yet, might be nice to have in a dev branch)

Reply:
Hey @karlb, this fork which is the cel2 fork is only used to deprecate functionality.

It's deprecating support for legacy celo transactions CeloDynamicFee transactions and CeloDenominated transactions.

That's because the prior fork (celo legacy fork) supports all celo transactions.

So you're suggesting that it would make more sense to not support CeloDenominated transactions in the celo legacy fork and then later enable support for it in the cel2 fork? If so I think I agree

Reply:
I was a bit irritated by the reverse logic of returning something when it is *not* supported, which is the opposite of what happens in other places. But it does give a nice error message to the user in this case ("deprecated" vs "not supported" tx).

> So you're suggesting that it would make more sense to not support CeloDenominated transactions in the celo legacy fork and then later enable support for it in the cel2 fork? If so I think I agree

Yes, I would prefer that.

Reply:
>I was a bit irritated by the reverse logic of returning something when it is not supported, which is the opposite of what happens in other places. But it does give a nice error message to the user in this case ("deprecated" vs "not supported" tx).

This made sense to me given that deprecations are actually new functionality that we will add in future forks. Is there an alternative design that you have in mind?

Reply:
Instead of iterating over all forks, you could only call the latest fork and that fork would have to either include all tx types explicitly or delegate to a previous fork. That way, you could not include unsupported tx types instead of having to override them with a deprecated returns.

That said, I don't have a problem with the current implementation.

Reply:
Ok got you, thanks!